pmm-app/src/pmm-system-summary-app-panel/module.ts

```diff
@@ -42,24 +42,32 @@ export class PanelCtrl extends MetricsPanelCtrl {
         [].forEach.call(rightScrollbarContainers, container => container.setAttribute('style', 'display: none !important'));
     }
 
-	link($scope, elem, attrs) {
+	link($scope, elem) {
 		const frame = elem.find('iframe');
 		const panel = elem.find('div.panel-container');
         const panelContent = elem.find('div.panel-content');
-		const bgcolor = $scope.qanParams.theme === 'light' ? '#ffffff' : '#141414';
-		panel.css({
-			'background-color': bgcolor,
-			'border': 'none'
-		});
+        const bgcolor = $scope.qanParams.theme === 'light' ? '#ffffff' : '#141414';
+
+        panel.css({
+            'background-color': bgcolor,
+            'border': 'none'
+        });
+
         this.disableGrafanaPerfectScroll(elem);
-        frame.on('load', () => {
-            frame.contents().bind('DOMSubtreeModified', () => setTimeout(() => {
-                    const h = frame.contents().find('body').height() || 400;
-                    frame.height(`${h + 100}px`);
-                    panel.height(`${h + 150}px`);
-                    panelContent.height(`inherit`);
-                }, 100)
-            )
-        })
-	}
+
+        $scope.ctrl.calculatePanelHeight = () => {
```

Comment:
These methods look identical in all three files. What's your opinion about to move them in separate file (shared.ts or something) file. Is this already actual?